pkg/volume/csi/csi_attacher.go

```diff
@@ -339,9 +339,16 @@ func (c *csiAttacher) MountDevice(spec *volume.Spec, devicePath string, deviceMo
 
 	// Store volume metadata for UnmountDevice. Keep it around even if the
 	// driver does not support NodeStage, UnmountDevice still needs it.
-	if err = os.MkdirAll(deviceMountPath, 0750); err != nil {
-		return errors.New(log("attacher.MountDevice failed to create dir %#v:  %v", deviceMountPath, err))
+	// If target path already exists (e.g. pod rescheduled on same node after
+	// node crash scenario), then continue to execute and
+	// let corresponding CSI driver take further action.
+	if _, err := os.Lstat(deviceMountPath); err != nil {
```

Comment:
CSI spec says that staging path must be a directory. `os.MkdirAll` returns no error when the directory already exists, even on Windows. If it returns error, it must have encountered something else than a directory, perhaps a link. This is not a valid behavior.
/hold
for discussion in https://github.com/kubernetes/kubernetes/issues/119401#issuecomment-1664022830

Reply:
Sorry, I forgot about this PR.

/hold cancel

Can you please create something like `MkdirAllWithMount` (I am bad at names...) and put it to say `pkg/util/filesystem` with a proper comment about Windows behavior of mounts + MkdirAll from https://github.com/kubernetes/kubernetes/issues/119401#issuecomment-1683694853, so this Windows knowledge does not get lost?

Reply:
Addressed your comment by adding new function `MkdirAllWithPathCheck` in pkg/util/filesystem, that looks exactly like `os.MkdirAll` with windows specific handling added.